Field.php

```diff
@@ -16,11 +16,11 @@ class Field
     protected $height;
     protected $maxSteps;
     private $history = [];
-    private $generationCount = 1;
+    private $generationCount = 0;
 
     /**
      * Field constructor
-     * Constructs an empty field, which can be overwritten by the child constructor.
+     * Constructs an empty field, be invoking the function constructField.
```

Comment:
Documentation is ok, but this is usually not necessary. It explains what the method does. This is not per se bad, but in this case it is an information that is quite obvious when looking at the code.
Where this is good is when it explains the bigger concept of a method. But this makes only sense if it is a big method that needs some overall explanation.